src/porepy/applications/verification_setups/mandel_biot.py

```diff
@@ -0,0 +1,1569 @@
+
+    exact_sol: MandelExactSolution
+    """Exact solution object."""
+
+    mdg: pp.MixedDimensionalGrid
+    """Mixed-dimensional grid."""
+
+    params: dict
+    """Parameter dictionary of the verification setup."""
+
+    stress_keyword: str
+    """Keyword for the mechanical subproblem."""
+
+    time_manager: pp.TimeManager
+    """Time-stepping object."""
+
+    def bc_type_mechanics(self, sd: pp.Grid) -> pp.BoundaryConditionVectorial:
+        """Define type of boundary conditions.
+
+        Rollers on all sides except on the East side which is Neumann.
+
+        Parameters:
+            sd: Subdomain grid.
+
+        Returns:
+            Vectorial boundary condition representation.
+
+        """
+        # Inherit bc from parent class. This sets all bc faces as Dirichlet.
+        bc = super().bc_type_mechanics(sd=sd)
+
+        # Get boundary sides, retrieve data dict, and bc object
+        sides = self.domain_boundary_sides(sd)
+
+        # East side: Neumann
+        bc.is_neu[:, sides.east] = True
+        bc.is_dir[:, sides.east] = False
+
+        # West side: Roller
+        bc.is_neu[1, sides.west] = True
+        bc.is_dir[1, sides.west] = False
+
+        # North side: Roller
+        bc.is_neu[0, sides.north] = True
+        bc.is_dir[0, sides.north] = False
+
+        # South side: Roller
+        bc.is_neu[0, sides.south] = True
+        bc.is_dir[0, sides.south] = False
+
+        return bc
+
+    def time_dependent_bc_values_mechanics(
+        self, subdomains: list[pp.Grid]
+    ) -> np.ndarray:
+        """Boundary values for mechanics.
+
+        Parameters:
+            subdomains: List of subdomains on which to define boundary conditions.
+
+        Returns:
+            Array of boundary values. Only non-zero values are the ones associated to
+            the North side of the domain.
+
+        """
+        bc_vals = super().time_dependent_bc_values_mechanics(subdomains)
+
+        sd = subdomains[0]
+        sides = self.domain_boundary_sides(sd)
+        yf_north = sd.face_centers[1][sides.north]
+
+        t = self.time_manager.time
+        uy_north_bc = self.exact_sol.vertical_displacement_profile(yf_north, t)
+        bc_vals[1::2][sides.north] = uy_north_bc
+
+        return bc_vals
+
+
+class MandelBoundaryConditionsSinglePhaseFlow(mass.BoundaryConditionsSinglePhaseFlow):
+
+    domain_boundary_sides: Callable[[pp.Grid], pp.bounding_box.DomainSides]
+    """Utility method to access the indices of the sides of the domain."""
+
+    def bc_type_darcy(self, sd: pp.Grid) -> pp.BoundaryCondition:
+        """Define boundary condition types for the flow subproblem.
+
+        All sides are set as Neumann, except the East side, which is Dirichlet.
+
+        Parameters:
+            sd: Subdomain grid.
+
+        Returns:
+            Scalar boundary condition representation.
+
+        """
+        # Inherit from base class. All sides are set as Dirichlet.
+        bc = super().bc_type_darcy(sd)
```

Comment:
Isn't this simply pp.BoundaryCondition(sd, self.domain_boundary_sides(sd).east, "dir")?

Reply:
Beautiful.